### PR TITLE
add testing for status quo of Byte and Short toHexString

### PR DESCRIPTION
**Pull request by som-snytt**
~Avoid sign extension in conversion to int for
purposes of enrichment.~

~Similarly for toOctalString and toBinaryString.~

references scala/bug#10216

### CODE REVIEW

**Comment by som-snytt** [comment]
Mima rose to her feet in the overheated courtroom and though her throat had gone dry managed to say in a voice that sounded very far away to her own ear but which seized the amazement of everyone in attendance:

`extension method toHexString$extension(Short)java.lang.String in object scala.runtime.RichShort does not have a correspondent in other version!`

File: test/files/run/richs.scala [hunk 1]
```diff
@@ -40,11 +43,96 @@ object RichCharTest1 extends RichTest {
...
 // }
+
+object RichByteTest extends RichTest {
+  override def test() = {
+    val sixteen = 16.toByte
+    println(sixteen.toBinaryString) // should be "10000"
+    println(sixteen.toHexString)    // should be "10"
+    println(sixteen.toOctalString)  // should be "20"
+    val max = 0x7F.toByte
+    println(max.toBinaryString) // should be "10000"
```
Comment (lrytz) [inline test/files/run/richs.scala]: The "should be" comments from here down seem to be off. Fixing or removing them is fine with me.

Comment (som-snytt) [reply test/files/run/richs.scala]: if twitter hadn't disabled my account for no reason, I would have tweeted something like TFW lrytz calls you on your test code comments. I don't remember the context here, but I will attempt to re-engage. Also, I don't miss twitter or nytimes crossword. I kind of miss nytimes, which made me feel like I was a smart person. That's also the reason I pretend to understand scala, so I feel smarter than the java guy at work.

Comment (som-snytt) [reply test/files/run/richs.scala]: Removed extraneous test code comments in case someone reads them.

**Comment by som-snytt** [comment]
Requires conditional compilation!

**Comment by som-snytt** [comment]
I'll reopen if 2.14 or 3.2, whichever comes first.

**Comment by som-snytt** [comment]
See https://github.com/scala/scala-library-next/pull/49 which may or may not supersede this.

**Comment by som-snytt** [comment]
Reopening this for improved test.

**Review by SethTisue — APPROVED** [review]

**Comment by som-snytt** [comment]
I had changed the title to "Byte and Short stuck with sign-extended toHexString", but I lost internet around 3 a.m. thx comcast.

**Pull request merged**